lab-anthony/lib/client-pool.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const EventEmitter = require('events');
+const uuid = require('node-uuid');
+
+function registerClient(socket, clientpool) {
+  socket.chat = {};
+  socket.chat.nick = 'guest_'+ Math.floor(Math.random() * (100 - 1)) + 1;
+  socket.chat.id = 'user_' + uuid.v4();
+  clientpool.pool[socket.chat.id] = socket;
+}
+
+function registerClientListeners(socket, clientpool) {
+  socket.on('data', (data) => {
+    if ((data.toString()).indexOf('\\nick') > -1){
+      var firstNick = socket.chat.nick;
+      clientpool.emit('newnick', data, socket);
+      var changedName = firstNick +' changed nickname to '+ socket.chat.nick;
+      clientpool.emit('broadcast', changedName, socket);
+      return;
+    }
+    clientpool.emit('broadcast', data, socket);
+  });
+
+  socket.on('close', () => {
+    console.log(socket.chat.id + 'has disconnected.');
+    delete clientpool.pool[socket.chat.id];
+  });
+
+  socket.on('error', function(err) {
+    console.error('CLIENT ERROR: ', err.message);
+  });
+}
+
+const ClientPool = module.exports = function() {
```

Comment:
I would either handle module.exports at the very bottom or at the very top so that your code is easier to read. Here it's hidden in the middle of a line in the middle of the code.